pkg/util/cloudprovider.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package util
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	"github.com/DataDog/datadog-agent/pkg/util/alibaba"
+	"github.com/DataDog/datadog-agent/pkg/util/azure"
+	"github.com/DataDog/datadog-agent/pkg/util/ec2"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs"
+	"github.com/DataDog/datadog-agent/pkg/util/gce"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetHostname retrieve the host name for the Agent, trying to query these
+// environments/api, in order:
+// * AWS ECS/Fargate
+// * AWS EC2
+// * GCE
+// * Azure
+// * Alibaba
+func DetectCloudProvider() {
+	detectors := []func() (bool, string){
+		ecs.IsRunningOn,
+		ec2.IsRunningOn,
+		gce.IsRunningOn,
+		azure.IsRunningOn,
+		alibaba.IsRunningOn,
+	}
+
+	for _, cloudDetector := range detectors {
```

Comment:
I don't really like the fact that `IsRunningOn` does both detection and returns it's `CloudProviderName`. I feel like the following would be cleaner and simpler:
```go
switch{
case ecs.IsRunningOn():
  inventories.SetAgentMetadata(inventories.CloudProviderMetatadaName, ecs.CloudProviderName)
case ec2.IsRunningOn():
  inventories.SetAgentMetadata(inventories.CloudProviderMetatadaName, ec2.CloudProviderName)
...
}
```
WDYT ?

Reply:
I think the log line will be very useful when troubleshooting agent's flares and I would like to avoid having too much duplicate code.

I also was thinking that we might want to return different cloud provider name for 1 cloud provide depending on the services use in it at some point. Having the package return the final name gives us more flexibility I think.

WDYT ?